jupyterhub/tests/test_spawner.py

```diff
@@ -483,3 +483,48 @@ def options_from_form(form_data, spawner):
     for key, value in form_data.items():
         assert key in result
         assert result[key] == value
+
+
+def test_spawner_server(db):
+    spawner = new_spawner(db)
+    orm_spawner = orm.Spawner()
+    orm_server = orm.Server(base_url="/1/")
+    orm_spawner.server = orm_server
+    db.add(orm_spawner)
+    db.add(orm_server)
+    db.commit()
+    # initial: no orm_spawner
+    assert spawner.server is None
```

Comment:
Error on this line, ping me for review when ready! I didn't quickly figure out what was wrong with assumptions/logic etc.